pkg/web/apps/apps_service.go

```diff
@@ -109,3 +110,165 @@ func (a *appsService) BindingAppByApp(appId, name string) error {
 	// if is deleted so just reactive the existent app
 	return a.repository.UnDeleteAppByAppID(app.AppID)
 }
+
+func (a *appsService) checkDeviceWithThisVersion(sdkInfo *models.Device, hasVersion bool, hasApp bool) (bool, *models.Device, error) {
+	// Has no need to do the queries if has not an app or version in the DB with the SDK info
+	if !hasApp || !hasVersion {
+		return false, nil, nil
+	}
+
+	device, err := a.repository.GetDeviceByVersionAndAppID(sdkInfo.Version, sdkInfo.AppID)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+
+	return true, device, nil
+}
+
+func (a *appsService) checkVersionForThisApp(sdkInfo *models.Device, hasApp bool) (bool, *models.Version, error) {
+	// Has no need to do the queries if has not an app in the DB with the SDK info
+	if !hasApp {
+		return false, nil, nil
+	}
+
+	version, err := a.repository.GetVersionByAppIDAndVersion(sdkInfo.AppID, sdkInfo.Version)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, version, nil
+}
+
+func (a *appsService) checkAnyVersionOfAppForThisDevice(sdkInfo *models.Device, hasVersion bool, hasApp bool) (bool, *models.Device, error) {
+	// Has no need to do the queries if has not an app or version in the DB with the SDK info
+	if !hasApp || hasVersion {
+		return false, nil, nil
+	}
+
+	device, err := a.repository.GetDeviceByDeviceIDAndAppID(sdkInfo.DeviceID, sdkInfo.AppID)
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, device, nil
+}
+
+func (a *appsService) checkIfHasApp(sdkInfo *models.Device) (bool, *models.App, error) {
+	app, err := a.repository.GetAppByAppID(sdkInfo.AppID)
+
+	if err != nil && err != models.ErrNotFound {
+		return false, nil, nil
+	}
+	return true, app, nil
+}
+
```

Comment:
See that all is encapsulated in the local funcs.
It is really nice in Go we are able to return more than one value from the funcs.